### ISSUE TITLE
Add kubeadm support for Istio local test installation

### ISSUE THREAD

**Issue by m1093782566** [istio/istio#8167]
**Describe the feature request**

Many PRs are blocked due to test failures, I feel a bit bad about the contributing experience. I think reproducing a test environment is the first step to reduce the failures but I was blocked in creating a local test environment.

I tried minikube and has no luck so far because my VM does not support nested virtualization even if I have installed kvm.

I am a Kubernetes maintainer and many k8s contributors are using kubeadm which supports single-node mode as well.

**Describe alternatives you've considered**

To workaround the unsupported nested virtualization issues, I am working on deploy Istio cluster via kubeadm tool. I would like to contribute the startup scripts and hope can help people who have the same troubles.

**Additional context**

**Comment by m1093782566** [comment]
cc @costinm @gargnupur @hklai @JimmyCYJ

**Comment by m1093782566** [comment]
If no one has objection to this, I will send a PR...

**Comment by hklai** [comment]
The group is also working on a new test framework that runs all required istio components in memory without having to depend on k8s or gke. That should greatly improve testing/contributing experience.

However, that does not necessarily remove the need for testing with a cluster. So something like this can still be useful in the cases you described.

**Comment by sdake** [comment]
@m1093782566 - note I use kubeadm (on bare metal) for all my workflows.  You may (or may not) find these scripts helpful:

```
#!/bin/bash

set -xe

echo Deploying Kubernetes
sudo kubeadm init --config=$HOME/kubeadm.conf

echo Setting Credentials for kubectl access
mkdir -p $HOME/.kube
sudo -H cp /etc/kubernetes/admin.conf $HOME/.kube/config
sudo -H chown $(id -u):$(id -g) $HOME/.kube/config

kubectl taint nodes --all=true  node-role.kubernetes.io/master:NoSchedule-

echo Deploying Calico CNI
kubectl apply -f $HOME/calico.yaml

echo Deploying metallb
kubectl apply -f https://raw.githubusercontent.com/google/metallb/v0.3.1/manifests/metallb.yaml
kubectl apply -f $HOME/configmap.yaml
```

where `configmap.yaml` is:
```
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: metallb-system
  name: config
data:
  config: |
    address-pools:
    - name: my-ip-space
      protocol: arp
      arp-network: 10.23.220.65/26
      cidr:
      - 10.23.220.90/30
```

and `kubeadm.conf` is:
```
apiVersion: kubeadm.k8s.io/v1alpha1
kind: MasterConfiguration
clusterName: falkor07
networking:
  serviceSubnet: 10.110.0.0/16
  podSubnet: 10.111.0.0/16
apiServerExtraArgs:
  runtime-config: admissionregistration.k8s.io/v1alpha1
```

Note I test with multicluster, which is the rationale behind changing the service and pod subnets.  If you don't want to do that, you can leave those out.   Also I think the runtime-config is no longer needed in kubernetes 1.9+.

My calico.yaml is calico upstream modified to network my pod and service subnets correctly.

metallb is a bare metal load balancer.

Cheers
-steve

**Comment by m1093782566** [comment]
thanks @sdake - it is truely helpful to my work. I will add these scripts into my PR :)

**Comment by rshriram** [comment]
@m1093782566 can you change our circleci scripts to deploy kubeadm instead of localkube? This will make sure that the kubeadm setup is well tested, and help you accurately reproduce CI failure locally as well. 

All the circleci configs are in .circleci/.. We are currently launching localkube for our e2e tests. Would love to have kubeadm there instead.

**Comment by m1093782566** [comment]
Sure @rshriram 

I am happy to join the work and it would be great if you can provide more context.

**Comment by m1093782566** [comment]
I just created a new issue for tracking this task, xref: #8365
